### PR TITLE
roadriot, roadriota: Dump comm board program

### PR DESCRIPTION
**Pull request by Hydreigon223**
Add the ADSP-2105 program from the hard to find com-link board only used in kit versions of Road Riot.

### CODE REVIEW

**Comment by Hydreigon223** [comment]
This board is exceedingly hard to find. The reason is because Atari only sold this board as an optional addon not for free instead of coming supplied with the conversion kit. It's likely back in the day operators found this addon board pointless if most only bought only one kit with no future plans to link a second kit. Many Road Riot boards I seen sold in the arcade community come without a com-link board.

File: src/mame/atari/atarig42.cpp [hunk 1]
```diff
@@ -11,10 +11,17 @@
         * Danger Express (1992)
         * Guardians of the 'Hood (1992)
 
-    Known bugs:
+    Note: Road Riot 4WD sports a communication link with another pcb allowing two players to
+    to compete with each other. The way both pcbs communicate depends on the program version.
+        * The kit version use an optional com-link board consisting on an ADSP-2105 (10Mhz XTAL)
+          using the ADSP's serial port
+        * The dedicated twin cab version uses dual port ram to bridge the two pcbs together (not
+          networked).
```
Comment (cuavas) [inline src/mame/atari/atarig42.cpp:19]: “Hertz” is a proper noun.

File: src/mame/atari/atarig42.cpp [hunk 2]
```diff
@@ -990,8 +1003,8 @@ void atarig42_0x400_state::init_guardian()
...
-GAME( 1991, roadriota, roadriot, atarig42_0x200, roadriot, atarig42_0x200_state, init_roadriot, ROT0, "Atari Games", "Road Riot 4WD (set 2, 13 Nov 1991)", MACHINE_UNEMULATED_PROTECTION | MACHINE_NODEVICE_LAN )
-GAME( 1991, roadriotb, roadriot, atarig42_0x200, roadriot, atarig42_0x200_state, init_roadriot, ROT0, "Atari Games", "Road Riot 4WD (set 3, 04 Jun 1991)", MACHINE_UNEMULATED_PROTECTION | MACHINE_NODEVICE_LAN )
+GAME( 1991, roadriot,  0,        atarig42_0x200, roadriot, atarig42_0x200_state, init_roadriot, ROT0, "Atari Games", "Road Riot 4WD (04 Dec 1991, Conversion Kit)", MACHINE_UNEMULATED_PROTECTION | MACHINE_NODEVICE_LAN )
+GAME( 1991, roadriota, roadriot, atarig42_0x200, roadriot, atarig42_0x200_state, init_roadriot, ROT0, "Atari Games", "Road Riot 4WD (13 Nov 1991, Conversion Kit)", MACHINE_UNEMULATED_PROTECTION | MACHINE_NODEVICE_LAN )
+GAME( 1991, roadriotb, roadriot, atarig42_0x200, roadriot, atarig42_0x200_state, init_roadriot, ROT0, "Atari Games", "Road Riot 4WD (04 Jun 1991, Dedicated Twin)", MACHINE_UNEMULATED_PROTECTION | MACHINE_NODEVICE_LAN )
```
Comment (cuavas) [inline src/mame/atari/atarig42.cpp:1008]: Don’t use uppercase for things that aren’t proper nouns in descriptive text.

File: src/mame/atari/atarig42.cpp (hunk 1 above)
Comment (Hydreigon223) [reply src/mame/atari/atarig42.cpp]: Note to self, capitalize the 'H'.

File: src/mame/atari/atarig42.cpp (hunk 2 above)
Comment (Hydreigon223) [reply src/mame/atari/atarig42.cpp]: Made changes like so.

File: src/mame/atari/atarig42.cpp [hunk 3]
```diff
@@ -11,10 +11,17 @@
...
-    Known bugs:
-        * ASIC65 for Road Riot not quite perfect
-        * Missing DSPCOM board for Road Riot 4WD
-            +or shared ram pcb that bridges both pcbs for the twin cab kind of like in F1: Exhaust Note and Air Rescue from segas32.cpp
+    Note: Road Riot 4WD sports a communication link with another pcb allowing two players to
+    to compete with each other. The way both pcbs communicate depends on the program version.
+        * The kit version uses an optional com-link board consisting on an ADSP-2105 (10MHz XTAL)
+          routing game link data through the ADSP's serial port.
+        * The dedicated twin cab version uses dual port ram to bridge the two pcbs together (not
+          networked).
```
Comment (angelosa) [inline src/mame/atari/atarig42.cpp:19]: `PCB`, uppercase. Otherwise the PR looks good to me.

Comment (cuavas) [reply src/mame/atari/atarig42.cpp:19]: “RAM” is also an acronym that should be capitalised.

Is the comment actually correct, though?  In particular, is there a reference for it using the ADSP-2105’s serial port?  @Hydreigon223 has been known to make incorrect assumptions about these things.

Saying “not networked” doesn’t make a great deal of sense, either.

Comment (Hydreigon223) [reply src/mame/atari/atarig42.cpp:19]: > “RAM” is also an acronym that should be capitalised.
> 
> Is the comment actually correct, though? In particular, is there a reference for it using the ADSP-2105’s serial port? @Hydreigon223 has been known to make incorrect assumptions about these things.
> 
> Saying “not networked” doesn’t make a great deal of sense, either.

Going to make the capitalization changes in a few minutes. As for the serial port, I'm going to trace the com-link board connector signals to see where they go to. Stay tuned.

**Pull request merged**